src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -334,14 +335,24 @@ public override bool IsBindingRequired
         {
             get
             {
-                if (MetadataKind == ModelMetadataKind.Property)
-                {
-                    return BindingMetadata.IsBindingRequired;
-                }
-                else
+                if (!_isBindingRequired.HasValue)
                 {
-                    return false;
+                    if (MetadataKind == ModelMetadataKind.Type)
+                    {
+                        _isBindingRequired = false;
```

Comment:
So a metadata provider that _wants_ to set `IsBindingRequired` for a `Type` can't? Realize the old code was doing something similar (ignoring the lack of parameter support, 'natch). But I'm wondering if it would be better to have this logic in our metadata providers rather than ignore what customer providers may do...